omod-common/src/main/java/org/openmrs/module/webservices/docs/swagger/SwaggerSpecificationCreator.java

```diff
@@ -319,13 +319,13 @@ private boolean testOperationImplemented(OperationEnum operation, DelegatingReso
 					} else {
 						try {
 							// to avoid saving data to the database, we pass a null SimpleObject
-							method.invoke(resourceHandler, null, RestConstants.SWAGGER_IMPOSSIBLE_UNIQUE_ID,
+							method.invoke(resourceHandler, RestConstants.SWAGGER_IMPOSSIBLE_UNIQUE_ID, null,
 							    new RequestContext());
 						}
-						catch (ResourceDoesNotSupportOperationException re) {
-							return false;
-						}
 						catch (Exception ee) {
+							if (ee.getCause() instanceof ResourceDoesNotSupportOperationException) {
```

Comment:
What is the exact class of exception when the cause is an instance of ResourceDoesNotSupportOperationException?

Reply:
InvocationTargetException

> its caused if the underlying method throws an exception.

Reply:
Can you put that as a comment in the code? That is because just looking at the same class caught a few lines above, one wonders why you would need to do this.